File: capacity-club-api/src/domain-modules/workout/model/payload/workout-create.payload.ts
import { ApiProperty } from '@nestjs/swagger';
import { IsString, Length, IsOptional, IsNotEmpty } from 'class-validator';
import { HomeTraining } from 'domain-modules/home-training/model/entity/home-training.entity';
import { IndoorTraining } from 'domain-modules/indoor-training/model/entity/indoor-training.entity';
import { TrainingCircuit } from 'domain-modules/training-circuit/model/entity/training-circuit.entity';

/**
 * Class used to create workout.
 * This class defines the payload used when creating new workout.
 */
export class WorkoutCreatePayload {
  @ApiProperty()
  @IsString()
  @IsNotEmpty()
  @Length(1, 50)
  title: string;
  @ApiProperty({ type: () => TrainingCircuit, isArray: true })
  @IsOptional()
  training_circuits: TrainingCircuit[];
  @ApiProperty({ type: () => HomeTraining })
  @IsOptional()
  home_training: HomeTraining;
  @ApiProperty({ type: () => IndoorTraining })
  @IsOptional()
  indoor_training: IndoorTraining;
}

File: capacity_club_mobile_app/lib/domain-features/data/workout/model/payload/workout_create_payload.dart
import 'package:capacity_club_mobile_app/domain-features/data/home-training/model/home_training_model.dart';
import 'package:capacity_club_mobile_app/domain-features/data/indoor-training/model/indoor_training_model.dart';
import 'package:capacity_club_mobile_app/domain-features/data/training-circuit/model/training_circuit_model.dart';
import 'package:equatable/equatable.dart';

class WorkoutCreatePayload extends Equatable {
  final String title;
  final List<TrainingCircuitModel> training_circuits;
  final IndoorTrainingModel indoor_training;
  final HomeTrainingModel home_training;

  const WorkoutCreatePayload({
    required this.title,
    required this.training_circuits,
    required this.indoor_training,
    required this.home_training,
  });

  Map<String, dynamic> toJson() => {
        'title': title,
        'training_circuits':
            training_circuits.map((item) => item.toJson()).toList(),
        'indoor_training': indoor_training.toJson(),
        'home_training': home_training.toJson(),
      };

  @override
  List<Object?> get props =>
      [title, training_circuits, indoor_training, home_training];
}
